articles/references/module_builder.md

```diff
@@ -0,0 +1,81 @@
+See [below](#model-type) for more info.
+
+
+## Template Settings
+
+Notice that when you click on a Template element in Intent Architect, that it will display these fields:
+
+![Template Element](images/TemplateElement.png)
+_Template Element_
+
+![Template Settings](images/TemplateSettings.png)
+_Template Settings_
+
+### Creation Mode
+The `Creation Mode` setting of a template determines what metadata format the template expects, and how files from a template must be created:
+1.	**Single File (No Model)** – create a single file without any additional metadata needed. A standalone C# class or ReadMe file would be examples of this template setting. 
+2.	**Single File (Model List)** – create a single file which takes in a list of metadata models. This is useful for creating classes which register other classes.
+3.	**File per Model** – create a file per metadata model. This is useful for creating a file based on a model such as a Domain class, Service, or DTO.
+4. **Custom** – you're on your own on this one. A minimalistic `Registration` class will be created whereby you will need to determine how many instances of that template will be created and from where.
+
+### Modeler
+The **Modeler** setting determines which _type_ of modeler the metadata should be fetched from for the template. This can be `Domain`, `Services`, `Eventing`, etc. When setting up your Module, the `Intent Module Builder` will automatically add the modeler dependency depending on which `Modeler` you have selected.
+
+### Model Type
+The **Model Type** setting specifies the object that you will be working with once the module code has been generated. You can select this Type based on `Types` that Intent Architect has been able to detect. More around this topic can be found [here](xref:IntentArchitectConcepts).
+
+### Expose Decorator Contract
+
+You might wish to create a Template that exposes an interface for a new Decorator to implement by applying the `Exposes Decorator Contract` Stereotype. The Type Fullname would be the C# Interface/class's full namespace and type name that will be used to allow Decorators to hook into this template.
+
+![Template Decorator Contract](images/TemplateDecoratorContract.png)
+
+## Decorator Settings
+
+Notice that when you click on a Decorator element in Intent Architect, that it will display these fields:
+
+![Decorator Element](images/DecoratorElement.png)
+_Decorator Element_
+
+![Decorator Settings](images/DecoratorSettings.png)
+_Decorator Settings_
+
+1. **Declare Usings** - When the Decorator class gets generated, allow the developer to specify the necessary "using" namespaces to be included upon decorator usage.
+2. **Type Fullname** - Give the FQDN of the desired class/interface that will be used to attach to.
```

Comment:
At least once above in this article we must have provided the definition of FQDN, if not, then this should link to it, or alternatively have a format like `FQDN (Fully Qualified D??? Name)`.